server/routes.js

```diff
@@ -17,6 +17,16 @@ app.get('/profile', ensureAuthenticated, function(req, res) {
     });
 });
 
+app.get('/profile-public/:id', ensureAuthenticated, function(req, res) {
+    User.findById(req.params.id, function(err, user) {
+        if (err) {
+            res.sendStatus(400);
+        } else {
+            res.json(user);
```

Comment:
At this point user variable may be `null` due to there is no user in DB with `req.params.id` which been provided - then server should respond with `404` status